agents/timestream_aggregator/timestream_aggregator.py

```diff
@@ -24,28 +30,319 @@ def _create_file_path(start_time, data_dir):
     Returns
     -------
     str
-        Full filepath of .g3 file for output
+        dirname for file for output
 
     """
-    start_datetime = datetime.datetime \
-                             .fromtimestamp(start_time,
-                                            tz=datetime
-                                            .timezone.utc)
-
     sub_dir = os.path.join(data_dir,
                            "{:.5}".format(str(start_time)))
 
     # Create new dir for current day
     if not os.path.exists(sub_dir):
         os.makedirs(sub_dir)
 
+    return sub_dir
+
+
+def _create_basename(start_time):
+    """Create the timestamp based basename for a file, i.e. 2019-01-01-13-37-00.
+
+    Arguments
+    ---------
+    start_time : float
+        Timestamp for start of data collection
+
+    Returns
+    -------
+    str
+        Basename for file. Suffix and extension to be added separately.
+
+    """
+    start_datetime = datetime.datetime \
+                             .fromtimestamp(start_time,
+                                            tz=datetime
+                                            .timezone.utc)
+
     time_string = start_datetime.strftime("%Y-%m-%d-%H-%M-%S")
-    filename = "{}.g3".format(time_string)
-    filepath = os.path.join(sub_dir, filename)
+
+    return time_string
+
+
+def _create_file_path(dirname, basename, suffix="", extension="g3"):
+    """Create the full path to the output file for writing.
+
+    Arguments
+    ---------
+    dirname : str
+        dirname for output, likely from _create_dirname
+    basename : str
+        basename of file without extension or suffix
+    suffix : str
+        Suffix to append to filename (include your own '_')
+    extension : str
+        file extension to put on file (without '.')
+
+    Returns
+    -------
+    str
+        Full filepath of file for output
+
+    """
+    filename = "{}{}.{}".format(basename, suffix, extension)
+    filepath = os.path.join(dirname, filename)
 
     return filepath
 
 
+class FrameRecorder:
```

Comment:
All file system operations and I/O are now isolated in this class; that's great.  I think you should emphasize somewhere that this should not be used in the reactor thread, for that reason.  Similary it would be good to annotate the relevant Process to indicate that it runs as a worker thread.  (Perhaps it is already -- I'm looking at a diff here...)

Reply:
Added a note about not running in the reactor to the docstring for the `FrameRecorder` class and for the `start_aggregation` process.